Add `FadeInImage.assetNetwork` example

swashcap opened the pull request:
* **Problem:** It's difficult to know how to use a bundled asset as a `FadeInImage` placeholder (see flutter/flutter#18333
* **Solution:** Add an example!
* **Testing:** ???

Closes flutter/flutter#18333.

swashcap commented:
Is there an easy way to make a GIF like the memory example? Or, should I just create a starter Flutter project and add the “complete example” code?

Hixie commented:
cc @Sfshaza

Sfshaza commented:
@swashcap, thanks for filling a hole in the cookbook!

Travis failed on the commit with this error:

  error • The argument type 'String' can't be assigned to the parameter type 'Uint8List' • example/fading_in_images_md_2.dart:21:13

The complete example looks fine to my naked eye, but something is unhappy.

@brianegan, please review and weigh in.

Sfshaza commented:
I just checked the fade-in example and, @brianegan, it's HUGE. Can it be reduced in size?

@swashcap, I don't know how Brian did it, but we have [these instructions on the Flutter wiki](https://docs.google.com/spreadsheets/d/1_mXxBKc9G9zCN1KNmUDSLrVoMFdFNnPGTtNofva775M/edit#gid=0).

brianegan commented on cookbook/images/fading-in-images.md:
```diff
@@ -68,3 +68,61 @@ class MyApp extends StatelessWidget {
+          title: new Text(title),
+        ),
+        body: Center(
+          child: new FadeInImage.memoryNetwork(
```
Hey there, thanks for the contribution!

It looks like this is the line that's causing the build to fail. I think if ya change this line to `FadeInImage.assetNetwork(...`, you should be all good!

swashcap replied:
Oh, whoops! Yep, I can do that.

swashcap replied:
Fixed in 38a1491a5ae846944c68daf020b33815a8361c2f.

brianegan commented:
@Sfshaza Yep! Looks like that one is crazy huge. I was told a redesign was coming soon and we'll have a better spot for content like this, but in the meantime I'll ensure we don't have any monsters like this one.

swashcap commented:
> It would be great to get rid of the `new` keyword from the complete example!

Removed in cbf18ea32e36a695624f4012de5e38e565825061.

I also added a demo GIF in c351351433b22f36906daffb66721db914307771! :tada:

brianegan commented:
Thanks for making those updates @swashcap. Looks good on my end!

Sfshaza commented:
I take that as an LGTM!

brianegan commented:
Thanks @Sfshaza :)
